src/matrix3.js

```diff
@@ -191,86 +173,56 @@ export default class Matrix3 extends MathArray {
 
   transformVector2(vector, out) {
     out = out || new Vector2();
-    vec2.transformMat4(out, vector, this);
-    // assert(validateVector2(out));
+    vec2.transformMat3(out, vector, this);
+    validateVector(out, 2);
     return out;
   }
 
-  transformVector3(vector, out = new Vector3()) {
+  transformVector3(vector, out) {
     out = out || new Vector3();
-    vec3.transformMat4(out, vector, this);
-    // assert(validateVector3(out));
+    vec3.transformMat3(out, vector, this);
+    validateVector(out, 3);
     return out;
   }
 
-  // transformVector4(vector, out = new Vector4()) {
-  //   out = out || new Vector4();
-  //   vec4.transformMat4(out, vector, this);
-  //   assert(validateVector4(out));
-  //   return out;
-  // }
-
-  // Transforms any 2, 3 or 4 element vector
-  // returns a newly minted Vector2, Vector3 or Vector4
   transformVector(vector, out) {
     switch (vector.length) {
       case 2:
         return this.transformVector2(vector, out);
       case 3:
         return this.transformVector3(vector, out);
-      // case 4:
-      //   return this.transformVector4(vector, out);
       default:
         throw new Error('Illegal vector');
     }
   }
 
   transformDirection(vector, out) {
+    return this._transformVector(vector, out, 0);
+  }
+
+  transformPoint(vector, out) {
+    return this._transformVector(vector, out, 1);
+  }
+
+  _transformVector(vector, out, w) {
     switch (vector.length) {
       case 2:
-        vec4.transformMat4(tempVector4, [vector[0], vector[1], 0, 0], this);
         out = out || new Vector2();
-        [out[0], out[1]] = tempVector4;
+        vec3.transformMat3(out, [vector[0], vector[1], w], this);
+        out.length = 2;
+        validateVector(out, 2);
         break;
       case 3:
-        vec4.transformMat4(tempVector4, [vector[0], vector[1], vector[2], 0], this);
+        if (Boolean(w) !== Boolean(vector[2])) {
+          throw new Error('math.gl: Matrix3.transformPoint - invalid vector');
+        }
         out = out || new Vector3();
-        [out[0], out[1], out[2]] = tempVector4;
+        vec3.transformMat3(out, vector, this);
+        validateVector(out, 3);
         break;
-      // case 4:
```

Comment:
Would could easily support vector4's. We would just treat this matrix as if it was a 4x4 matrix with this matrix as the upper left "minor": 

```
| mat3 0 |
| 0    1 |
```